rest/route/auth.go

```diff
@@ -0,0 +1,55 @@
+package route
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+
+	"github.com/evergreen-ci/gimlet"
+	"github.com/pkg/errors"
+)
+
+////////////////////////////////////////////////////////////////////////
+//
+// POST /rest/v2/auth
+
+type authPermissionGetHandler struct {
+	resource      string
+	resourceType  string
+	permission    string
+	requiredLevel int
+}
+
+func makeGetAuthPermissionHandler() gimlet.RouteHandler {
+	return &authPermissionGetHandler{}
+}
```

Comment:
I would avoid the constructor function and just manually declare the object in service.go